tests/tox.sh

```diff
@@ -24,70 +24,83 @@ EOF
 
 # Check distro and install deps
 if command -v apt-get &>/dev/null; then
-    sudo apt-get install -y --force-yes docker.io
-    sudo apt-get install -y --force-yes xfsprogs
-  else
-    sudo yum install -y docker
-    sudo yum install -y xfsprogs
-    if ! systemctl status docker >/dev/null; then
-      # daemon doesn't start automatically after being installed
-      sudo systemctl restart docker
-    fi
-    # Allow running `docker` without sudo
-    sudo chgrp "$(whoami)" /var/run/docker.sock
+  sudo apt-get install -y --force-yes docker.io xfsprogs python3.6
+  sudo ln -sf "$(command -v python3.6)" /usr/bin/python3
+else
+  sudo yum install -y docker xfsprogs
+  if ! command -v python3.6 &>/dev/null; then
+    sudo yum -y groupinstall development
+    sudo yum -y install https://centos7.iuscommunity.org/ius-release.rpm
+    sudo yum -y install python36u
+  fi
+  sudo ln -sf "$(command -v python3.6)" /usr/bin/python3
+
+  if ! systemctl status docker >/dev/null; then
+    # daemon doesn't start automatically after being installed
+    sudo systemctl restart docker
+  fi
+  # Allow running `docker` without sudo
+  sudo chgrp "$(whoami)" /var/run/docker.sock
 fi
 
 rm -rf "$WORKSPACE"/ceph-ansible || true
 git clone -b $CEPH_ANSIBLE_BRANCH --single-branch https://github.com/ceph/ceph-ansible.git ceph-ansible
 
-if [ "$CEPH_ANSIBLE_BRANCH" == 'stable-2.2' -o "$CEPH_ANSIBLE_BRANCH" == 'stable-3.0' ]; then
+if [[ "$CEPH_ANSIBLE_BRANCH" == 'stable-2.2' ]] || [[ "$CEPH_ANSIBLE_BRANCH" == 'stable-3.0' ]]; then
   REQUIREMENTS=requirements2.2.txt
 else
   REQUIREMENTS=requirements2.4.txt
 fi
 
 pip install -r $TOXINIDIR/ceph-ansible/tests/$REQUIREMENTS
 
-# pull requests tests should never have these directories here, but branches
-# do, so for the build scripts to work correctly, these neeed to be removed
-# XXX It requires sudo because these will appear with `root` ownership
-rm -rf "$WORKSPACE"/{daemon,demo,base}
-
 bash "$WORKSPACE"/travis-builds/purge_cluster.sh
 # XXX purge_cluster only stops containers, it doesn't really remove them so try to
 # remove them for real
 containers_to_remove=$(sudo docker ps -a -q)
 
 if [ "${containers_to_remove}" ]; then
-    sudo docker rm -f $@ ${containers_to_remove} || echo failed to remove containers
+  sudo docker rm -f "$@" ${containers_to_remove} || echo failed to remove containers
+fi
+
+cd "$WORKSPACE"
+
+# build everything that was touched to make sure build succeeds
+FLAVOR_ARRAY=( $(make flavors.modified) )
+
+if [[ "${#FLAVOR_ARRAY[@]}" -eq "1" ]]; then
+  FLAVOR="${FLAVOR_ARRAY[0]}"
+else
+  # if more than one release/distro is impacted then we test this in priority
+  FLAVOR="luminous,amd64,centos,7,_,centos,7"
```

Comment:
It would be more consistent to have ubuntu here instead of centos as we do tag it as latest.
This way, we have the latest which is the most tested.